procbuild/builder.py

```diff
@@ -77,99 +76,188 @@ def shell(cmd, path=None, retry=0):
+        else:
+            self.add_output('[*] Updating proceedings build tools in'
+                            ' {self.master_repo_path}...\n')
+            errcode, output = shell('git pull', self.master_repo_path, retry=2)
+        
+        self.add_output(output)
+        
+        if errcode:
+            self.add_output('[X] Error code %d ' % errcode)
+            raise BuildError("get_build_tools")
+        
+        self._remove_papers_dir()
+        
+    
+    def _remove_papers_dir(self):
+        """ this command will remove the papers dir from the build_tools dir
+        
+        """
+        errcode, output = shell('rm -rf papers', 
+                                path=self.master_repo_path)
+        
+        self.add_output(output)
+
+        if errcode:
+            self.add_output('[X] Error code %d ' % errcode)
+            raise BuildError("remove_papers_dir")
+
+    def _checkout_paper_repo(self):
+        self.add_output('[*] Check out paper repository...\n')
+        errcode, output = checkout(repo(self.user), 
+                                   self.branch, 
+                                   self.build_path)
+        self.add_output(output)
+        
+        if errcode:
+            self.add_output('[X] Error code %d\n' % errcode)
+            self.build_status = 'Failed to check out paper'
+            raise BuildError('checkout_paper_repo')
+
+    
+    def _relocate_build_tools(self):
+        """Move local build tools into temporary directory
+        """
+        self.add_output('Moving proceedings build tools to temp directory...\n')
+        errcode, output = shell('cp -r '
+                                f'{self.master_repo_path}/. {self.build_path}')
+        self.add_output(output)
+        if errcode:
+            self.add_output('[X] Error code %d\n' % errcode)
+            self.build_status = 'Could not move build tools to temp directory'
+            raise BuildError('relocate_build_tools')
+
+    def _relocate_static_files(self):
+        for f in self.data_files:
+            shutil.copy(f, self.paper_path)
+    
+    @property
+    def paper_path(self):
+        return joinp(self.build_path, 'papers', self.paper)
+        
+    @property
+    def paper(self):
+        papers = [p for p in iglob(self.build_path + '/papers/*')
+                  if not any(p.endswith(e) for e in excluded)]
+
+        if self.build_path is None:
+            self.add_output('[X] No build path declared: %s\n' % papers)
+            self.build_status = 'No build path declared'
+            raise BuildError('papers')
+        
+        if len(papers) < 1:
+            self.add_output('[X] No papers found: %s\n' % papers)
+            self.build_status = 'Paper not found'
+            raise BuildError('papers')
+        
+        else:
+            return papers[0].split('/')[-1]
+    
+    def _run_make_paper_script(self):
+        """Runs the scipy_proceedings make_paper.sh script
```

Comment:
Isn't this script just a wrapper around the build manager?

Reply:
No, this actually builds the paper using the make script from scipy_proceedings.

Reply:
I tried to write that in the docstring for that method… can you suggest how it could be made clearer?